test/e2e/framework/debug/dump.go

```diff
@@ -106,41 +110,93 @@ func dumpAllNodeInfo(ctx context.Context, c clientset.Interface, nodes *v1.NodeL
 // DumpNodeDebugInfo dumps debug information of the given nodes.
 func DumpNodeDebugInfo(ctx context.Context, c clientset.Interface, nodeNames []string, logFunc func(fmt string, args ...interface{})) {
 	for _, n := range nodeNames {
-		logFunc("\nLogging node info for node %v", n)
+		logFunc("")
+		logFunc("Logging node info for node %v", n)
 		node, err := c.CoreV1().Nodes().Get(ctx, n, metav1.GetOptions{})
 		if err != nil {
 			logFunc("Error getting node info %v", err)
 		}
-		logFunc("Node Info: %v", node)
-
-		logFunc("\nLogging kubelet events for node %v", n)
+		nodeString, _ := json.Marshal(node)
```

Comment:
I recently enhanced gomega.Format so that falls back to YAML marshalling for API objects: https://github.com/kubernetes/kubernetes/pull/113384/commits/5a01a52b0cd6afaf2d1508bfa8ad38febea5c0d8

Is that perhaps more readable here?

```suggestion
		nodeString, _ := format.Object(node)
```

`format` == `k8s.io/kubernetes/test/utils/format`.

Reply:
It could.  I think there's a tension in this entire block of output:

1. It's printed on every e2e failure, even those where nodes aren't relevant
2. It's *mostly* an artifact of historical status of nodes being out of date
3. Current output is ~20 lines wrapped, but output isn't easily parseable later by a human (<- my primary motivation for this change)
4. We want to maximize information density and give you enough info to quickly debug 80-90% of obvious node/pod failures
5. The more nodes in a cluster the less useful this section is (right now we have a flag to have a cutoff), when *really* we want to highlight the nodes running the pods from the current test.

If we want to print the whole thing, JSON is at least more scannable than now and more compact than a full yaml doc (which would be many more lines), so the ratio of utility/lines drops with yaml.  However, we *could* potentially elide parts of the node object.  We could also pick which parts of the info to show (a healthy node needs less output, maybe cap).

Opinions?

Reply:
Rule of thumb is you want to only include info that can be relevant.  Events are high value, but need to be correlated with other timelines (various projects have attempted this over the years, openshift does a ton of post processing to create a single timeline of all relevant transitions which i eventually want to move upstream).

I share your intuition that we could potentially track what the test does and guide what we show based on that.  We don’t want much info, or too complicated heuristics, but within that we could probably hit the mark (show nodes if you created pods on those nodes, show all nodes if your test called ListNodes).

I am trying to improve the actual output concretely in small directions, so maybe what we need to do is document some guidelines (and red lines… like don’t remove anything a particular test definitely needs?) and seek convergence?